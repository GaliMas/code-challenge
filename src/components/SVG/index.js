import React from 'react';
import PropTypes from 'prop-types';

const SVG = ({ id }) => (
  <svg>
    <use xlinkHref={`./sprite.svg#${id}`} />
  </svg>
);

SVG.propTypes = {
  id: PropTypes.string,
};

export default SVG;
