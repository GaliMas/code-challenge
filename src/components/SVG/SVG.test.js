import React from 'react';
import renderer from 'react-test-renderer';
import SVG from '.';

it('renders correctly', () => {
  const tree = renderer.create(
    <div>
      <SVG id="billin-usage" />
      <SVG id="github-usage" />
      <SVG id="linkedin" />
      <SVG id="twitter" />
    </div>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
