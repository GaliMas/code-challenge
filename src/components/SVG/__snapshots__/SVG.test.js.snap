exports[`test renders correctly 1`] = `
<div>
  <svg>
    <use
      xlinkHref="./sprite.svg#billin-usage" />
  </svg>
  <svg>
    <use
      xlinkHref="./sprite.svg#github-usage" />
  </svg>
  <svg>
    <use
      xlinkHref="./sprite.svg#linkedin" />
  </svg>
  <svg>
    <use
      xlinkHref="./sprite.svg#twitter" />
  </svg>
</div>
`;
